### PR TITLE
Add S3_AccessControl hook

### PR DESCRIPTION
**Pull request by mrinaudo-aws**
*Issue #, if available:*

*Description of changes:*
Add S3_AccessControl hook.

## Unit tests excerpts
```
[...]
[INFO] Running com.awscommunity.s3.accesscontrol.PreUpdateHookHandlerTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.354 s -- in com.awscommunity.s3.accesscontrol.PreUpdateHookHandlerTest
[INFO] Running com.awscommunity.s3.accesscontrol.PreCreateHookHandlerTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.025 s -- in com.awscommunity.s3.accesscontrol.PreCreateHookHandlerTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0
[...]
[INFO] --- jacoco:0.8.10:check (jacoco-check) @ awscommunity-s3-accesscontrol-handler ---
[...]
[INFO] Analyzed bundle 'awscommunity-s3-accesscontrol-handler' with 5 classes
[INFO] All coverage checks have been met.
```

## Contract tests excerpts
```
[...]
5 passed, 2 skipped, 17 deselected
[...]
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: hooks/S3_AccessControl/README.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+For more information, see [Testing resource types locally using AWS SAM](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test.html#resource-type-develop-test).
+
+Open a another terminal window, build this hook, and run contract tests:
+
+```shell
+cfn generate && mvn clean verify && cfn test -v --enforce-timeout 90
+```
+
+## Hook development notes
+The RPDK will automatically generate the correct hook input model from the schema whenever the project is built via Maven. You can also do this manually with the following command: `cfn generate`.
```
Comment (ericzbeard) [inline hooks/S3_AccessControl/README.md:143]: I don't think "RPDK" is a public term.

File: hooks/S3_AccessControl/README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            "FailureMode": "FAIL",
+            "Properties": {
+            }
+        }
+    }
+}
+EOF
+```
+
+and then submit the hook configuration using the [AWS Command Line Interface (AWS CLI)](https://aws.amazon.com/cli/).  First, get the [Amazon Resource Name](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (ARN) for this hook, as follows (examples are for the `us-east-1` region):
```
Comment (ericzbeard) [inline hooks/S3_AccessControl/README.md:45]: Replace double spaces after periods with single space. :)

**Review by ericzbeard — CHANGES_REQUESTED** [review]
Very thorough as always! Only a few minor comments.

Are you going to add this to the pipeline in a separate PR?

File: hooks/S3_AccessControl/README.md (hunk 1 above)
Comment (mrinaudo-aws) [reply hooks/S3_AccessControl/README.md:143]: I see it described [here](https://github.com/aws-cloudformation/cloudformation-cli-python-plugin#aws-cloudformation-resource-provider-python-plugin-1).  That initial line is actually created by the cfn-cli tool in the README file when you create a new extension project.  I have added lines below for this specific Kotlin project, and left the original line intact.

File: hooks/S3_AccessControl/README.md (hunk 2 above)
Comment (mrinaudo-aws) [reply hooks/S3_AccessControl/README.md:45]: Will do - thank you!

File: hooks/S3_AccessControl/README.md (hunk 1 above)
Comment (mrinaudo-aws) [reply hooks/S3_AccessControl/README.md:143]: I will update this "Hook development notes" section.

**Comment by mrinaudo-aws** [comment]
> Very thorough as always! Only a few minor comments.
> 
> Are you going to add this to the pipeline in a separate PR?

I can add an updated pipeline template here as well.

**Review by ericzbeard — APPROVED** [review]

**Pull request merged**